chirp/drivers/icr6.py

```diff
@@ -0,0 +1,794 @@
+  u8 unknown7:2,
+     ctone:6;             // TSQL Index. Valid range: 0 to 49
+  u8 unknown8;
+  u8 canceller_freq_h;    // Canceller training freq index - high 8 bits
+  u8 canceller_freq_l:1,  //                               - LSB
+     unknown10m:4,
+     vsc:1,               // Voice Squelch Control: 0=Off, 1=On
+     canceller:2;         // USA-only Canceller option: Index to "CANCELLER"
+  u8 name0;
+  u8 name1;
+  u8 name2;
+  u8 name3;
+  u8 name4;
+} memory[1300];
+
+// Unknown: 0x5140 to 0x5F7F inclusive
+struct {
+  u8 unknown0;
+  u8 unknown1;
+  u8 unknown2;
+  u8 unknown3;
+  u8 unknown4;
+  u8 unknown5;
+  u8 unknown6;
+  u8 unknown7;
+  u8 unknown8;
+  u8 unknown9;
+  u8 unknown10;
+  u8 unknown11;
+  u8 unknown12;
+  u8 unknown13;
+  u8 unknown14;
+  u8 unknown15;
+} mystery1[200];
+
+// 25x Scan Edges with names - 0x5dc0 to 0x5f4f inclusive
+// Mulitply the 32-bit by 3 to get freq in Hz
+struct {
+  u8 start0;      // LSB
+  u8 start1;
+  u8 start2;
+  u8 start3;      // MSB
+  u8 end0;        // LSB
+  u8 end1;
+  u8 end2;
+  u8 end3;        // MSB
+  u8 disabled:1,
+     mode:3,      // 0=FM, 1=WFM, 2=AM, 4="-"
+     ts:4;        // Same mapping as channel TS
+  u8 unknown9a:2,
+     attn:2,      // 0=Off, 1=On, 2="-"
+     unknown9b:4;
+  char name[6];
+} pgmscanedge[25];
+
+// Unknown
+struct {
+  u8 unknown0;
+  u8 unknown1;
+  u8 unknown2;
+  u8 unknown3;
+  u8 unknown4;
+  u8 unknown5;
+  u8 unknown6;
+  u8 unknown7;
+  u8 unknown8;
+  u8 unknown9;
+  u8 unknown10;
+  u8 unknown11;
+  u8 unknown12;
+  u8 unknown13;
+  u8 unknown14;
+  u8 unknown15;
+} mystery1b[3];
+
+// Channel control flags: 0x5F80 to 0x69A7 inclusive
+struct {
+  u8 hide_channel:1,  // Channel enable/disable aka show/hide
+     skip:2,          // Scan skip: 0=No, 1 = "Skip", 3 = mem&vfo ("P")
+     unknown0d:1,
+     unknown0e:1,
+     unknown0f:1,
+     unknown0g:1,
+     unknown0h:1;
+  u8 unknown1;
+} flags[1300];
+
+// Eight bytes of Padding,  everything seems to work on 16-byte boundaries:
+struct {
+  u8 unknown0; // Start: 0x69A8
+  u8 unknown1; // End:   0x69AF
+  u8 unknown2;
+  u8 unknown3;
+  u8 unknown4;
+  u8 unknown5;
+  u8 unknown6;
+  u8 unknown7;
+} padding;
+
+// Unknown: Coded as 34 blocks of 16-bytes. 0x69B0 to 0x6BCF
+struct {
+  u8 unknown0;
+  u8 unknown1;
+  u8 unknown2;
+  u8 unknown3;
+  u8 unknown4;
+  u8 unknown5;
+  u8 unknown6;
+  u8 unknown7;
+  u8 unknown8;
+  u8 unknown9;
+  u8 unknown10;
+  u8 unknown11;
+  u8 unknown12;
+  u8 unknown13;
+  u8 unknown14;
+  u8 unknown15;
+} mystery2[34];
+
+// Device Settings: 0x6bd0 to 0x6c0f
+struct {
+  u8 unknown00;  // 6bd0
+  u8 unknown01;  // 6bd1
+  u8 unknown02;  // 6bd2
+  u8 unknown03;  // 6bd3
+  u8 unknown04;  // 6bd4
+  u8 unknown05;  // 6bd5
+  u8 unknown06;  // 6bd6
+  u8 unknown07;  // 6bd7
+  u8 unknown08;  // 6bd8
+  u8 unknown09;  // 6bd9
+  u8 unknown10;  // 6bda
+  u8 unknown11;  // 6bdb
+  u8 unknown12;  // 6bdc
```

Comment:
It'd be a lot more readable if you'd just change this to `u8 unknown[13]` and that's definitely the convention.

Reply:
Fixed